uaclient/cli.py

```diff
@@ -346,28 +346,48 @@ def _attach_with_token(
     return 0
 
 
-@assert_not_attached
-@assert_root
-def action_attach_premium(args, cfg):
+def _get_contract_token_from_cloud_identity(cfg: config.UAConfig) -> str:
+    """Detect cloud_type and request a contract token from identity info.
+
+    :param cfg: Parsed configuration dictionary
+
+    :raise NonPremiumImageError: When not on a premium image type. Exits 0.
```

Comment:
I'm not sure that the exit code is relevant here; callers may catch the exception and discard it, which would mean we wouldn't exit 0.

Reply:
Looking at this now.

Reply:
Fix pushed.